src/cmd/flux-jobs.py

```diff
@@ -0,0 +1,202 @@
+    parser.add_argument("-A", action="store_true", help="List all jobs for all users")
+    parser.add_argument(
+        "-c",
+        "--count",
+        type=int,
+        metavar="N",
+        default=1000,
+        help="Limit output to N jobs(default 1000)",
+    )
+    parser.add_argument(
+        "-s",
+        "--states",
+        type=str,
+        metavar="STATES",
+        default="pending,running",
+        help="List jobs in specific states(pending,running,inactive)",
+    )
+    parser.add_argument(
+        "--suppress-header",
+        action="store_true",
+        help="Suppress printing of header line",
+    )
+    parser.add_argument(
+        "-u",
+        "--userid",
+        type=int,
+        metavar="UID",
+        default=os.geteuid(),
+        help='Limit output to specific userid (Specify "all" for all users)',
+    )
+    parser.add_argument(
+        "-o",
+        "--format",
+        type=str,
+        metavar="FORMAT",
+        help="Specify output format using Python f-string formation",
+    )
+
+    args = parser.parse_args()
+
+    h = flux.Flux()
+
+    # Future optimization, reduce attrs based on what is in output
+    # format, to reduce potentially large return RPC.
+    attrs = [
+        "userid",
+        "priority",
+        "state",
+        "job-name",
+        "task-count",
+        "t_submit",
+        "t_depend",
+        "t_sched",
+        "t_run",
+        "t_cleanup",
+        "t_inactive",
+    ]
+
+    if args.a:
+        args.userid = os.geteuid()
+    if args.A:
+        args.userid = flux.constants.FLUX_USERID_UNKNOWN
+    if args.a or args.A:
+        args.states = "pending,running,inactive"
+
+    flags = 0
+    for state in args.states.split(","):
+        if state.lower() == "pending":
+            flags |= flux.constants.FLUX_JOB_LIST_PENDING
+        elif state.lower() == "running":
+            flags |= flux.constants.FLUX_JOB_LIST_RUNNING
+        elif state.lower() == "inactive":
+            flags |= flux.constants.FLUX_JOB_LIST_INACTIVE
+        else:
+            print("Invalid state specified", file=sys.stderr)
+            sys.exit(1)
+
+    if flags == 0:
+        flags |= flux.constants.FLUX_JOB_LIST_PENDING
+        flags |= flux.constants.FLUX_JOB_LIST_RUNNING
+
+    rpc_handle = flux.job.job_list(h, args.count, attrs, args.userid, flags)
+    try:
+        jobs = flux.job.job_list_get(rpc_handle)
+    except EnvironmentError as e:
+        print("{}: {}".format("rpc", e.strerror), file=sys.stderr)
+        sys.exit(1)
+
+    if args.format:
+        output_format(args.format, jobs)
+    else:
+        if not args.suppress_header:
+            print(
+                "%s\t\t%-8.8s %-6.6s %-8.8s %-8.8s %s"
+                % ("JOBID", "STATE", "TASKS", "NAME", "USER", "TIME")
+            )
+        fmt = (
+            "{id}\t{state:<8.8} {task_count:<6} "
+            "{job_name:<8.8} {username:<8.8} {runtime_fsd}"
```

Comment:
The default format could use some tweaking IMO. This is very subjective, but at the least:

 * more space for job name (since there is currently extra width available) Maybe 10 characters at least
 * It would be nice if runtime was blank instead of "0s" for "pending" jobs. Actually, maybe TIME could be "time since submission" for pending jobs? (Not sure if that is a good idea or not)
 * I think JOBID is going to need more than "2 tabs" worth of spaces by default. O/w after some amount of time FLUIDs will overflow this space and the header won't line up. Probably won't be an issue in the short term, but I fast forwarded FLUID generator by six days and got the output below. Probably want to use a default width for `{id}` of 18. (Techinically I guess we'd need 20, but by my testing it takes about 1.88 years for the FLUIDs to roll over to 19 digits)
```
$ flux jobs -a
JOBID		STATE    TASKS  NAME     USER     TIME
8697698811117568	INACTIVE 10     hostname grondo   0.08s
8697690472841216	INACTIVE 10     hostname grondo   0.08s
8697681698357248	INACTIVE 10     hostname grondo   0.08s
8697674333159424	INACTIVE 10     hostname grondo   0.08s
8697374104879104	INACTIVE 10     sleep    grondo   10s
```

* It is weird for me to see the username as one of the last fields in the default output format. Again this is totally subjective, but maybe the default format should be `JOBID  USER  NAME STATE NTASKS TIME`.

Reply:
> It is weird for me to see the username as one of the last fields in the default output format.

FWIW, the order I picked was somewhat random :-)  Re-ordered 1 or 2 times b/c a certain order "felt" good.  But we can totally re-order. :-)